arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2893,6 +2920,319 @@ void RestReplicationHandler::handleCommandLoggerTickRanges() {
   }
 }
 
+bool RestReplicationHandler::prepareRevisionOperation(RevisionOperationContext& ctx) {
+  auto& selector = server().getFeature<EngineSelectorFeature>();
+  if (!selector.isRocksDB()) {
+    generateError(
+        rest::ResponseCode::NOT_IMPLEMENTED, TRI_ERROR_NOT_IMPLEMENTED,
+        "this storage engine does not support revision-based replication");
+    return false;
+  }
+
+  LOG_TOPIC("253e2", TRACE, arangodb::Logger::REPLICATION)
+      << "enter prepareRevisionOperation";
+
+  bool found = false;
+
+  // get collection Name
+  ctx.cname = _request->value("collection");
+  if (ctx.cname.empty()) {
+    generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "invalid collection parameter");
+    return false;
+  }
+
+  // get batchId
+  std::string const& batchIdString = _request->value("batchId", found);
+  if (found) {
+    ctx.batchId = StringUtils::uint64(batchIdString);
+  } else {
+    generateError(rest::ResponseCode::NOT_FOUND, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "replication revision tree - request misses batchId");
+    return false;
+  }
+
+  // get resume
+  std::string const& resumeString = _request->value("resume", found);
+  if (found) {
+    ctx.resume = StringUtils::uint64(resumeString);
+  }
+
+  // print request
+  LOG_TOPIC("2b70f", TRACE, arangodb::Logger::REPLICATION)
+      << "requested revision tree for collection '" << ctx.cname
+      << "' using contextId '" << ctx.batchId << "'";
+
+  ExecContextSuperuserScope escope(ExecContext::current().isAdminUser());
+
+  if (!ExecContext::current().canUseCollection(_vocbase.name(), ctx.cname, auth::Level::RO)) {
```

Comment:
do we need to special handle _users here ?

Reply:
I am not aware whether `_users` requires special handling here, in general. Why would `canUseCollection` not work?

The current implementation side-steps this question by excluding system collections from eligibility for this syncing protocol. That may change in the future, so it would be good to figure this out now.